### ISSUE TITLE
 cucumber-cpp conan create error

### ISSUE THREAD

**Issue by willcode4food** [ThoSe1990/cucumber_example#1]
Hello,

Thank you for providing such an informative example of using cucumber.  I'm running into an error when building cucumber-cpp for conan, `conan create ./conanfile/cucumber-cpp/conanfile.py cucumber-cpp/0.5@cwt/stable --build missing`.   Any insight you could provide to a beginner at modern C++ development would be much appreciated.

I am using an Ubuntu docker container to work with the repo and I am getting the following error. I'm not sure where to being to debug.  I have the dependencies (except QT) installed and available. Below is the stack trace and the Dockerfile I use to build the image I run the dev container from.  Please let me know if there is any other information I should provide. 

```
/home/cppdev/.conan/data/boost/1.79.0/_/_/package/adf48b8e4446dcb68f440fe42ff08878d5c7feed/include/boost/multi_array.hpp:231:49: error: implicitly-declared ‘constexpr boost::multi_array_ref<std::basic_string<char>, 2>::multi_array_ref(const boost::multi_array_ref<std::basic_string<char>, 2>&)’ is deprecated [-Werror=deprecated-copy]
231 | alloc_base(static_cast<const alloc_base&>(rhs)) {
| ^
In file included from /home/cppdev/.conan/data/boost/1.79.0/_/_/package/adf48b8e4446dcb68f440fe42ff08878d5c7feed/include/boost/multi_array.hpp:34,
from /home/cppdev/.conan/data/cucumber-cpp/0.5/cwt/stable/build/5f5fdaf06f8bd18961521dbe87621377ba962aa5/source_subfolder/include/cucumber-cpp/internal/connectors/wire/../../CukeEngine.hpp:8,
from /home/cppdev/.conan/data/cucumber-cpp/0.5/cwt/stable/build/5f5fdaf06f8bd18961521dbe87621377ba962aa5/source_subfolder/include/cucumber-cpp/internal/connectors/wire/WireProtocol.hpp:6,
from /home/cppdev/.conan/data/cucumber-cpp/0.5/cwt/stable/build/5f5fdaf06f8bd18961521dbe87621377ba962aa5/source_subfolder/include/cucumber-cpp/internal/connectors/wire/WireProtocolCommands.hpp:4,
from /home/cppdev/.conan/data/cucumber-cpp/0.5/cwt/stable/build/5f5fdaf06f8bd18961521dbe87621377ba962aa5/source_subfolder/src/connectors/wire/WireProtocolCommands.cpp:1:
/home/cppdev/.conan/data/boost/1.79.0/_/_/package/adf48b8e4446dcb68f440fe42ff08878d5c7feed/include/boost/multi_array/multi_array_ref.hpp:490:20: note: because ‘boost::multi_array_ref<std::basic_string<char>, 2>’ has user-provided ‘boost::multi_array_ref<T, NumDims>& boost::multi_array_ref<T, NumDims>::operator=(const boost::multi_array_ref<T, NumDims>&) [with T = std::basic_string<char>; long unsigned int NumDims = 2]’
490 | multi_array_ref& operator=(const multi_array_ref& other) {
| ^~~~~~~~
[ 88%] Building CXX object source_subfolder/src/CMakeFiles/cucumber-cpp.dir/drivers/GTestDriver.cpp.o

...

cucumber-cpp/0.5@cwt/stable: ERROR: Package '5f5fdaf06f8bd18961521dbe87621377ba962aa5' build failed
cucumber-cpp/0.5@cwt/stable: WARN: Build folder /home/cppdev/.conan/data/cucumber-cpp/0.5/cwt/stable/build/5f5fdaf06f8bd18961521dbe87621377ba962aa5
ERROR: cucumber-cpp/0.5@cwt/stable: Error in build() method, line 85
        cmake.build(target="cucumber-cpp")
        ConanException: Error 2 while executing cmake --build '/home/cppdev/.conan/data/cucumber-cpp/0.5/cwt/stable/build/5f5fdaf06f8bd18961521dbe87621377ba962aa5' '--target' 'cucumber-cpp' '--' '-j4'
```

The docker container I am working with is built with the following dockerfile

```
FROM ubuntu:focal

ARG DEBIAN_FRONTEND=noninteractive

# Install base packages
RUN set -ex && \
    apt-get update && \
    apt-get install -y lsb-release wget software-properties-common

# Install clang toolchain version 14
RUN wget https://apt.llvm.org/llvm.sh &&\
    chmod +x llvm.sh && \
    ./llvm.sh 14

# Install c++ development libraries
RUN set -ex && \
    apt-get update && \
    apt-get install -y build-essential \
    cmake \
    gdb \
    zsh \
    curl \
    gdb \
    libncurses-dev \
    flex \
    bison \
    gperf \
    python3 \
    python3-venv \
    ninja-build \ 
    ccache \
    libffi-dev \
    libssl-dev \
    dfu-util \
    libusb-1.0-0 \
    libboost-all-dev \
    pip

RUN pip install conan

RUN chsh -s $(which zsh) 
```

**Comment by ThoSe1990** [comment]
Hi there and thanks for the feedback

TL;DR I added a Dockerfile file to this directory `./docker/Dockerfile` which just worked fine for me.


I just gave it a try with your docker and unfortunately the error occurs while building the cucumber sources in WireProtocolCommands.cpp by using boost. To me this seems like a bug on cucumber's site. I don't know with which compilers they are testing and which boost version they use. I tried on your debian boost 1.79, 1.80 and 1.81 and non worked. Maybe there is also an problem with the gcc 9.2 which is used by default on the debian. 

So maybe this is solved on the debian system by upgrading gcc to 11.x or 12.x. 

I also noticed that by default installing docker it installs conan 2.0. I haven't had the time to try 2.0 but it seems that some commands are broken. So make sure you run pip install conan==1.57.0

Anyway, I added my ubuntu docker which I use. So to get it running right now to play around, I'd suggest using just this one. 

I also updated the readme slightl.

You can let me know if the docker works out for you.

Best Thomas

**Comment by willcode4food** [comment]
Thank you Thomas.  You gave me some things work with.  I'll give them a try.

**Comment by willcode4food** [comment]
@ThoSe1990 switching from ubuntu:20.04 (focal) to 22.02 (jammy) seems to have resolved the issue.  It's using gcc 11.3 and pinned to conan 1.57.  Thanks!
